web/src/message_view.js

```diff
@@ -642,7 +655,18 @@ export function show(raw_terms, opts) {
                 }
                 message_fetch.load_messages_for_narrow({
                     anchor,
+                    validate_filter_topic_post_fetch:
+                        filter.requires_adjustment_for_moved_with_target,
                     cont() {
+                        if (
+                            !filter.requires_adjustment_for_moved_with_target &&
+                            filter.has_operator("with")
+                        ) {
+                            // In this situation, we've adjusted our filter via
+                            // filter.try_adjusting_for_moved_with_target, and should
+                            // update the URL hash accordingly.
+                            update_hash_to_match_filter(filter, "retarget topic location");
```

Comment:
It seems to me that in the event that the `with` operator is for a message in the expected view, this will cause an unnecessary duplicate call to `changehash`, since `try_adjusting_for_moved_with_target` sets `filter.requires_adjustment_for_moved_with_target=false` regardless of whether there were any terms to adjust.

Is this a bug, or is the intent that this operation should do something, like remove the `with` clause from the URL?

Reply:
Opened https://github.com/zulip/zulip/issues/30862 to track this detail.

Reply:
it definitely seems like a bug. the very initial version of PR used to changehash only if the narrow terms were updated. I guess it somehow got removed, I'll try to update it.